lib/index.js

```diff
@@ -122,149 +229,23 @@ var util = require('./common/util.js');
  * //-
  */
 function gcloud(config) {
-  return {
-    bigquery: function(options) {
-      options = options || {};
-      return new BigQuery(util.extendGlobalConfig(config, options));
-    },
-    datastore: new Datastore(config),
-    dns: function(options) {
-      options = options || {};
-      return new DNS(util.extendGlobalConfig(config, options));
-    },
-    pubsub: function(options) {
-      options = options || {};
-      return new PubSub(util.extendGlobalConfig(config, options));
-    },
-    search: function(options) {
-      options = options || {};
-      return new Search(util.extendGlobalConfig(config, options));
-    },
-    storage: function(options) {
-      options = options || {};
-      return new Storage(util.extendGlobalConfig(config, options));
+  return Object.keys(apis).reduce(function(gcloudExposedApi, apiName) {
+    var Class = apis[apiName];
+
+    if (scopedApis[apiName]) {
+      gcloudExposedApi[apiName] = new Class(config);
+    } else {
+      gcloudExposedApi[apiName] = function(options) {
+        return new Class(util.extendGlobalConfig(config, options));
+      };
```

Comment:
I can revert to the old way if you prefer. With this way of doing things, I'm only bothered by the odd-ball Datastore instantiation we have, but I actually think this helps explain it and adds documentation around it. If you don't merge by tomorrow, I'll take the hint and go back to the old way :)